### ISSUE TITLE
User story - Social login/registration

### ISSUE THREAD

**Issue by stephendawsondev** [stephendawsondev/StackPortfolio#13]
### If applicable, add a related Epic.

Epic: #5 

### User Story

As a **user**, I can **do sign up or log in using a social account** so that I can **easily signup and have my details prepopulated**.

### Acceptance Criteria

- The user should be able to sign up with LinkedIn, Gmail or GitHub
- The user should be able to sign in with their selected social account

### Please ensure you've completed these tasks before marking the issue as done.

- [x] Reviewed the user story for clarity and completeness.
- [x] Identified and linked related issues or epics.
- [x] Created or updated the necessary documentation.
- [ ] Written unit and/or integration tests for functionality.
- [ ] All code changes have been manually tested to ensure they meet the acceptance criteria.
- [x] Updated the issue with any relevant notes or feedback (to date).
- [ ] Merged into the main branch and deployed to production (if applicable).

### Please add any further tasks that need to be completed before the issue can be marked as done.

- [x] Look into allauth to see if it can provide social login functionality

**Comment by stephendawsondev** [comment]
It turns out that this version of allauth may have issues with authenticating with LinkedIn: https://github.com/pennersr/django-allauth/issues/1641

I'll leave that out for now and just add Github authentication.

**Comment by stephendawsondev** [comment]
This has been added in #90 . I'll close this for now, but may come back to add automated tests (somehow) 😅 In any case, I'll be carrying out manual tests.
